forcejson/scanner_test.go

```diff
@@ -10,9 +6,156 @@ import (
+			}
+
+			buf.Reset()
+			if err := Compact(&buf, []byte(tt.indent)); err != nil {
+				GinkgoT().Errorf("Compact(%#q): %v", tt.indent, err)
+				continue
+			} else if s := buf.String(); s != tt.compact {
+				GinkgoT().Errorf("Compact(%#q) = %#q, want %#q", tt.indent, s, tt.compact)
+			}
+		}
+	})
+	It("compact separators", func() {
+
+		tests := []struct {
+			in, compact string
+		}{
+			{"{\"\u2028\": 1}", `{"\u2028":1}`},
+			{"{\"\u2029\" :2}", `{"\u2029":2}`},
+		}
+		for _, tt := range tests {
+			var buf bytes.Buffer
+			if err := Compact(&buf, []byte(tt.in)); err != nil {
+				GinkgoT().Errorf("Compact(%q): %v", tt.in, err)
+			} else if s := buf.String(); s != tt.compact {
+				GinkgoT().Errorf("Compact(%q) = %q, want %q", tt.in, s, tt.compact)
+			}
+		}
+	})
+	It("indent", func() {
+
+		var buf bytes.Buffer
+		for _, tt := range examples {
+			buf.Reset()
+			if err := Indent(&buf, []byte(tt.indent), "", "\t"); err != nil {
+				GinkgoT().Errorf("Indent(%#q): %v", tt.indent, err)
+			} else if s := buf.String(); s != tt.indent {
+				GinkgoT().Errorf("Indent(%#q) = %#q, want original", tt.indent, s)
+			}
+
+			buf.Reset()
+			if err := Indent(&buf, []byte(tt.compact), "", "\t"); err != nil {
+				GinkgoT().Errorf("Indent(%#q): %v", tt.compact, err)
+				continue
+			} else if s := buf.String(); s != tt.indent {
+				GinkgoT().Errorf("Indent(%#q) = %#q, want %#q", tt.compact, s, tt.indent)
+			}
+		}
+	})
+	It("compact big", func() {
+
+		initBig()
+		var buf bytes.Buffer
+		if err := Compact(&buf, jsonBig); err != nil {
+			GinkgoT().Fatalf("Compact: %v", err)
+		}
+		b := buf.Bytes()
+		if !bytes.Equal(b, jsonBig) {
+			GinkgoT().Error("Compact(jsonBig) != jsonBig")
+			diff(GinkgoT(), b, jsonBig)
+			return
+		}
+	})
+	It("indent big", func() {
+
+		initBig()
+		var buf bytes.Buffer
+		if err := Indent(&buf, jsonBig, "", "\t"); err != nil {
+			GinkgoT().Fatalf("Indent1: %v", err)
+		}
+		b := buf.Bytes()
+		if len(b) == len(jsonBig) {
+			GinkgoT().Fatalf("Indent(jsonBig) did not get bigger")
+		}
+
+		var buf1 bytes.Buffer
+		if err := Indent(&buf1, b, "", "\t"); err != nil {
+			GinkgoT().Fatalf("Indent2: %v", err)
+		}
+		b1 := buf1.Bytes()
+		if !bytes.Equal(b1, b) {
+			GinkgoT().Error("Indent(Indent(jsonBig)) != Indent(jsonBig)")
+			diff(GinkgoT(), b1, b)
+			return
+		}
+
+		buf1.Reset()
+		if err := Compact(&buf1, b); err != nil {
+			GinkgoT().Fatalf("Compact: %v", err)
+		}
+		b1 = buf1.Bytes()
+		if !bytes.Equal(b1, jsonBig) {
+			GinkgoT().Error("Compact(Indent(jsonBig)) != jsonBig")
+			diff(GinkgoT(), b1, jsonBig)
+			return
+		}
+	})
+	It("indent errors", func() {
+
+		for i, tt := range indentErrorTests {
+			slice := make([]uint8, 0)
+			buf := bytes.NewBuffer(slice)
+			if err := Indent(buf, []uint8(tt.in), "", ""); err != nil {
+				if !reflect.DeepEqual(err, tt.err) {
+					GinkgoT().Errorf("#%d: Indent: %#v", i, err)
+					continue
+				}
+			}
+		}
+	})
+	It("next value big", func() {
```

Comment:
nit: this is cool but results in some janky non-human descriptions like these. Maybe we want these to be better if we're to maintain them

Reply:
Agreed, just seems outside the scope of this change. Let's go with the fix-when-touching tactic. Fair?